internal/storage/auth/keycloak/user.go

```diff
@@ -121,6 +122,27 @@ func (r *UserRepository) GetAll(ctx context.Context) ([]*entities.User, error) {
 	return allUsers, nil
 }
 
+func (r *UserRepository) GetAllByRole(ctx context.Context, role entities.UserRole) ([]*entities.User, error) {
+	log := logger.GetLogger(ctx)
+	users, err := r.GetAll(ctx)
+	if err != nil {
+		log.Debug("failed to fetch all users", "error", err)
+		return nil, service.MapError(ctx, err, service.ErrorLogEntityNotFound)
+	}
+
+	var filteredUsers []*entities.User
+	for _, user := range users {
+		for _, userRole := range user.Roles {
+			if userRole == role {
+				filteredUsers = append(filteredUsers, user)
+				break
+			}
+		}
+	}
+
+	return filteredUsers, nil
```

Comment:
You could use some standard iterator functions such as filter, which are defined in the utils package (`utils.Filter`), to reduce the ugly for loops, but this is up to you :man_shrugging: